### PR TITLE
WIP:Shellcheck

### PR DESCRIPTION
**Pull request by Ingi-Hong**
Reference issue #211 

WIP
install.ps1
install.sh
readme

Working
releases.js

### CODE REVIEW

**Comment by coolaj86** [comment]
@Ingi-Hong Go ahead and start on the README. I'll review the code today or Saturday.

**Comment by coolaj86** [comment]
Seems to install correctly on my Mac, and on Linux. Checking Windows momentarily...

I suggest that you run these scripts through both [`shfmt`](https://webinstall.dev/shfmt) (using the options specified in [`vim-shfmt`](https://webinstall.dev/vim-shfmt) and `shellcheck` (how apropos).

This is the format I recommend for the version function.

```bash
shellcheck --version | head -n 2 | tail -n 1 | cut -d' ' -f 2
```

**Comment by coolaj86** [comment]
Here's what's needed for the Windows releases:

```diff
diff --git a/shellcheck/releases.js b/shellcheck/releases.js
index 902200e..124b52f 100644
--- a/shellcheck/releases.js
+++ b/shellcheck/releases.js
@@ -6,6 +6,16 @@ var repo = 'shellcheck';

 module.exports = function (request) {
   return github(request, owner, repo).then(function (all) {
+    all.releases.forEach(function (rel) {
+      // if there is no os or arch or source designation, and it's a .zip, it's Windows amd64
+      if (
+        !/(darwin|mac|linux|x86_64|arm|src|source)/i.test(rel.name) &&
+        /\.zip$/.test(rel.name)
+      ) {
+        rel.os = 'windows';
+        rel.arch = 'amd64';
+      }
+    });
     return all;
   });
 };
```

File: shellcheck/install.ps1 [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    # TODO: create package-specific temp directory
+    # Enter tmp
+    pushd .local\tmp
+
+        # Remove any leftover tmp cruft
+        Remove-Item -Path ".\shellcheck-v*" -Recurse -ErrorAction Ignore
+        Remove-Item -Path ".\shellcheck.exe" -Recurse -ErrorAction Ignore
+
+        # NOTE: DELETE THIS COMMENT IF NOT USED
```
Comment (coolaj86) [inline shellcheck/install.ps1:41]: delete this, it's not used

File: shellcheck/install.ps1 [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        # Unpack archive file into this temporary directory
+        # Windows BSD-tar handles zip. Imagine that.
+        echo "Unpacking $pkg_download"
+        & tar xf "$pkg_download"
+
+        # Settle unpacked archive into place
+        echo "Install Location: $pkg_src_cmd"
+        New-Item "$pkg_src_bin" -ItemType Directory -Force
+        Move-Item -Path ".\shellcheck-*\shellcheck.exe" -Destination "$pkg_src_bin"
```
Comment (coolaj86) [inline shellcheck/install.ps1:53]: This unzips without any container directory. This should work:

```pwsh
Move-Item -Path ".\shellcheck*.exe" -Destination "$pkg_src_bin"
```

**Comment by coolaj86** [comment]
I updated my diff for release.js above because this repo also uses tags for `latest` and `stable` which need to be filtered out because they interfere with the normal version selection.

(and we already tag `latest` and `stable` anyway by inspecting the version and channel info)

**Comment by Ingi-Hong** [comment]
Thanks for taking the time to help me out! 

From my understanding, git --diff just shows the difference between to files? So everything with a + in releases.js is what you want me to add.

**Comment by coolaj86** [comment]
This looks right. I'll test and then one last thing:

1. Do a rebase
    ```bash
    git rebase -i HEAD~5
    ```
2. mark your first commit as `reword` (you'll give it a new name, such as "add shellcheck")
3. mark every commit afterwards as `fixup`
4. force push
    ```bash
    git push --force <your origin> <your branch>
    ```

**Review by coolaj86 — APPROVED** [review]

**Pull request merged**